js/ui/appActivation.js

```diff
@@ -27,6 +27,40 @@ const SPLASH_SCREEN_TIMEOUT = 700; // ms
 const DEFAULT_MAXIMIZED_WINDOW_SIZE = 0.75;
 const LAUNCH_MAXIMIZED_DESKTOP_KEY = 'X-Endless-LaunchMaximized';
 
+const LaunchReason = {
+    UNINTERESTING: 0,
+    CODING_BUILDER: 1
+};
+
+
+// Determine if a splash screen should be shown for the provided
+// GDesktopAppInfo, LaunchReason and and other global settings
+function _shouldShowSplash(info, launchReason) {
```

Comment:
That function should be:

```
diff --git a/js/ui/appActivation.js b/js/ui/appActivation.js
index 3161385..e46b799 100644
--- a/js/ui/appActivation.js
+++ b/js/ui/appActivation.js
@@ -35,7 +35,9 @@ const LaunchReason = {
 
 // Determine if a splash screen should be shown for the provided
 // GDesktopAppInfo, LaunchReason and and other global settings
-function _shouldShowSplash(info, launchReason) {
+function _shouldShowSplash(app, launchReason) {
+    let info = app.get_app_info();
+
     // Short-circuit - if we're activating GNOME-Builder for Coding,
     // then we always want to show a splash screen
     if (launchReason === LaunchReason.CODING_BUILDER) {
@@ -55,10 +57,12 @@ function _shouldShowSplash(info, launchReason) {
     // Don't show splash screen if this is a link and the browser is
     // running. We can't rely on any signal being emitted in that
     // case, as links open in browser tabs.
-    if (this._app.get_id().indexOf('eos-link-') != -1 &&
+    if (app.get_id().indexOf('eos-link-') != -1 &&
         Util.getBrowserApp().state != Shell.AppState.STOPPED) {
         return false;
     }
+
+    return true;
 }
 
 const AppActivationContext = new Lang.Class({
@@ -119,7 +123,7 @@ const AppActivationContext = new Lang.Class({
     },
 
     showSplash: function(launchReason=LaunchReason.UNINTERESTING) {
-        if (!_shouldShowSplash(this._app.get_app_info(), launchReason)) {
+        if (!_shouldShowSplash(this._app, launchReason)) {
             return;
         }

We need to return true as default here otherwise no splash screen is shown for applications like our Browser.
```